assignment_2/lca.py

```diff
@@ -0,0 +1,70 @@
+from math import sqrt
+
+class Lca_finder:
+    # we divide levels of tree into zones (groups of levels with that have hegiht of sqrt(height)
```

Comment:
As a reader I would appreciate a bit more explanation about what the levels and zones mean. Maybe a diagram? For the one writing the code it may be obvious but a reader needs to make an effort to figure it out.

More generally, having read this explanation I do not know what algorithm the class implements and do not know what to expect below. It might also help if the functions below had documentation or comments.